src/hypercore-file-logger.js

```diff
@@ -86,47 +89,62 @@ class HyperCoreFileLogger extends HyperCoreLogger {
     return prefix + line
   }
 
-  async start () {
-    const files = await resolvePaths(this.pathlike)
-    if (!files.length) throw new Error('ERR_FILE_NOT_FOUND')
+  /**
+   * @private
+   */
+  async watchFile (file, miltiple = false) {
+    if (this.republish) {
+      const encoding = this.feedOpts.valueEncoding
+      const rstream = fs.createReadStream(file, { encoding })
+
+      const rl = readline.createInterface({
+        input: rstream,
+        crlfDelay: Infinity
+      })
 
-    await super.start()
+      for await (const line of rl) {
+        await new Promise((resolve, reject) => {
+          const data = HyperCoreFileLogger.formatLine(line, miltiple && file) + '\n'
+          this.feed.append(data, (err) => err ? reject(err) : resolve())
+        })
+      }
+    }
 
-    await Promise.all(files.map(async (file) => {
-      if (this.republish) {
-        const encoding = this.feedOpts.valueEncoding
-        const rstream = fs.createReadStream(file, { encoding })
+    const tail = new Tail(file)
+    tail.on('line', (line) => {
+      const data = HyperCoreFileLogger.formatLine(line, miltiple && file) + '\n'
+      this.feed.append(data)
+    })
+    tail.watch()
 
-        const rl = readline.createInterface({
-          input: rstream,
-          crlfDelay: Infinity
-        })
+    this.fileTails[file] = tail
+  }
 
-        for await (const line of rl) {
-          await new Promise((resolve, reject) => {
-            const data = HyperCoreFileLogger.formatLine(line, files.length > 1 && file) + '\n'
-            this.feed.append(data, (err) => err ? reject(err) : resolve())
-          })
-        }
-      }
+  /**
+   * @private
+   */
+  async unwatchFile (file) {
+    const watcher = this.fileTails[file]
 
-      const tail = new Tail(file)
-      tail.on('line', (line) => {
-        const data = HyperCoreFileLogger.formatLine(line, files.length > 1 && file) + '\n'
-        this.feed.append(data)
-      })
-      tail.watch()
+    if (watcher) {
+      watcher.unwatch()
+      delete this.fileTails[file]
+    }
+  }
 
-      this.fileTails[file] = tail
-    }))
+  async start () {
+    await super.start()
+    this.watcher = chokidar.watch(this.pathlike)
 
-    debug('feed started listening for changes on %s', files.join(', '))
+    this.watcher.on('add', file => this.watchFile(file, isGlob(this.pathlike)))
```

Comment:
question, if we create a new file e.g. like echo 'test' > 123.log, does it get published through hypercore even if republish tag is missing? if content is added later I assume it should work due to tail functionality

Reply:
> does it get published through hypercore even if republish tag is missing

nope

Reply:
ok so let's see this case:
**writer**
```
node bin/hyperlog.js write --file ../logs/
```
**reader**
```
node bin/hyperlog.js read --key 079936557d2fa792d5729686d368f35c624320babcf2345a53fc0b55b259661e
```

currently since there's no republish functionality old content won't get published, then let's say we publish new content to existing file, after that the content is retrieved in log which is ok

then let's consider this case:
```
echo 'test 33' > ../logs/log-3.txt
```
In case above log-3 doesn't exist before so it's created new and the content is added to test 33, in this case the log is not red from reader, afaik it should be red since this is a new added content after started listening to changes

Reply:
done